lib/SILOptimizer/Utils/SILInliner.cpp

```diff
@@ -559,15 +559,18 @@ SILValue SILInlineCloner::borrowFunctionArgument(SILValue callArg,
       || callArg.getOwnershipKind() != ValueOwnershipKind::Owned) {
     return callArg;
   }
-  auto *borrow = getBuilder().createBeginBorrow(AI.getLoc(), callArg);
+
+  SILBuilderWithScope beginBuilder(AI.getInstruction());
```

Comment:
I was just thinking about this. Should we pass down the SILBuilderContext here?

Reply:
My take is that creating a `SILBuilderWithScope` is _extremely_ cheap so, given we don't really know whether the instructions down will have the same scope of the `apply`, we might as well create a fresh object.

Reply:
The reason I am asking is there might be code that is passing in an InsertedInstrs
or something like that.

Reply:
I might be wrong here so please correct me, but I think this doesn't really matter here. We create new `SILBuilderWithScopes` on instructions that aren't inserted.

Reply:
Resolved this in person & changed the code to pass a `SILBuilderContext` around.